site-src/concepts/guidelines.md

```diff
@@ -38,6 +38,28 @@ prescribed resolution, the following guiding principles should be applied:
   should be given to the resource appearing first in alphabetical order
   (namespace/name). For example, foo/bar would be given precedence over foo/baz.
 
+## Gracefully Handling Future API Versions
+
+An important consideration when implementing this API is how it might change in
+the future. Similar to the Ingress API before it, this API is designed to be
+implemented by a variety of different products within the same cluster. That
+means that the API version your implementation was developed with may be
+different than the API version it is used with. At a minimum, the following
+requirements must be met to ensure future versions of the API do not break your
+implementation:
+
+* Handle fields with loosened validation without crashing
+* Handle fields that have transitioned from required to optional without
+  crashing
+* DO NOT rely on webhook or CRD validation as a security mechanism. If field
```

Comment:
What is "escaping" referring to in the webhook? We don't transform any field values?

Reply:
I've removed this comment, but for example our CRD validation uses regexes to ensure that only "safe" values will exist for some fields. That does not mean that implementations should blindly trust this input and add it to a DB query for example.